### PR TITLE
feat: Support Build on M1 MacOS

### PR DESCRIPTION
**Pull request by GehaFearless**
issue: https://github.com/apache/incubator-pegasus/issues/1093

patch relate by https://github.com/jemalloc/jemalloc/issues/1997

### CODE REVIEW

**Comment by neverchanje** [comment]
```
CMake Error at /opt/homebrew/Cellar/cmake/3.23.3/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the
  system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY
  OPENSSL_INCLUDE_DIR)
```

Did you encounter this problem? Testing on my local mac m1.

**Comment by GehaFearless** [comment]
> ```
> CMake Error at /opt/homebrew/Cellar/cmake/3.23.3/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
>   Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the
>   system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY
>   OPENSSL_INCLUDE_DIR)
> ```
> 
> Did you encounter this problem? Testing your branch on my local mac m1.

You should `export OPENSSL_ROOT_DIR="/usr/local/opt/openssl"`.
Also you need `brew install snappy
          brew install lz4
          brew install zstd
          brew install openssl@1.1`

**Comment by acelyc111** [comment]
> > Did you encounter this problem? Testing your branch on my local mac m1.
> 
> You should `export OPENSSL_ROOT_DIR="/usr/local/opt/openssl"`. Also you need `brew install snappy brew install lz4 brew install zstd brew install openssl@1.1`

Yes, reference to https://github.com/apache/incubator-pegasus/blob/master/.github/workflows/lint_and_test_cpp.yaml#L298

File: .licenserc.yaml [hunk 1]
```diff
@@ -90,6 +90,7 @@ header:
     - 'thirdparty/fix_libevent_for_macos.patch'
     - 'thirdparty/fix_s2_for_aarch64.patch'
     - 'thirdparty/fix_thrift_for_cpp11.patch'
+    - 'thirdparty/fix_jemalloc_for_m1_on_macos.patch'
```
Comment (acelyc111) [inline .licenserc.yaml:93]:
```suggestion
    - 'thirdparty/fix_jemalloc_for_m1_on_macos.patch'
    - 'thirdparty/fix_libevent_for_macos.patch'
    - 'thirdparty/fix_s2_for_aarch64.patch'
    - 'thirdparty/fix_thrift_for_cpp11.patch'
```
keep in lexicographic order

**Comment by acelyc111** [comment]
could you check why jemalloc hasn't been rebuilt?
https://github.com/apache/incubator-pegasus/runs/7624482151?check_suite_focus=true#step:5:31

**Comment by GehaFearless** [comment]
> Details

https://github.com/apache/incubator-pegasus/runs/7644893971?check_suite_focus=true

**Review by acelyc111 — APPROVED** [review]

**Review by neverchanje — APPROVED** [review]
Haven't tried yet but if it won't impact the linux build then we can merge it now.

**Pull request merged**